File: server/src/main/com/broll/gainea/server/core/objects/monster/GodDragon.java
package com.broll.gainea.server.core.objects.monster;

import com.broll.gainea.server.core.battle.BattleResult;

public class GodDragon extends Monster {

    public GodDragon() {
        setIcon(58);
        setName("Götterdrache");
        setStats(8, 8);
        setBehavior(MonsterBehavior.AGGRESSIVE);
        setActivity(MonsterActivity.ALWAYS);
        setMotion(MonsterMotion.AIRBORNE);
    }

    @Override
    public void onDeath(BattleResult throughBattle) {
        if (throughBattle != null) {
            throughBattle.getKillingPlayers(this).forEach(p -> p.getGoalHandler().addPoints(1));
        }
    }
}

File: server/src/main/com/broll/gainea/server/core/objects/monster/GodDragon.kt
package com.broll.gainea.server.core.objects.monster

import com.broll.gainea.server.core.battle.BattleResult
import com.broll.gainea.server.core.player.Player

class GodDragon(owner: Player) : Monster(owner) {
    init {
        icon = 58
        name = "Götterdrache"
        setStats(8, 8)
        behavior = MonsterBehavior.AGGRESSIVE
        activity = MonsterActivity.ALWAYS
        motion = MonsterMotion.AIRBORNE
        description = "Bezwinger erhält einen Siegespunkt"
    }

    override fun onDeath(throughBattle: BattleResult?) {
        throughBattle?.let {
            it.getKillingPlayers(this).forEach { player -> player.goalHandler.addPoints(1) }
        }
    }
}
